### PR TITLE
Login

### PR DESCRIPTION
**Pull request by Allegraknox**
Adds log in features so user can sign up, sign in, edit profile, and delete profile. Does not allow duplicate emails, password must be 6 letters.

### CODE REVIEW

**Review by mrubakh — APPROVED** [review]

File: marketplaceApp/app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        @user = User.new
+    end
+    
+    def create
+        @user = User.new(user_params)
+        if @user.save
+            flash[:notice] = "New user #{@user.email} created"
+            redirect_to root_path and return
+        else
+            flash[:alert] = "Failed to save new product"
```
Comment (amycsong98) [inline marketplaceApp/app/controllers/users_controller.rb:12]: Failed to save new user

File: marketplaceApp/app/views/users/registrations/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    <%= f.label :current_password %> <i>(we need your current password to confirm your changes)</i><br />
+    <%= f.password_field :current_password, autocomplete: "current-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Update" %>
+  </div>
+<% end %>
+
+<h3>Cancel my account</h3>
```
Comment (amycsong98) [inline marketplaceApp/app/views/users/registrations/edit.html.erb:49]: what does cancel mean though? a bit ambiguous

Comment (amycsong98) [reply marketplaceApp/app/views/users/registrations/edit.html.erb:49]: `Delete` my account? maybe.

File: marketplaceApp/db/seeds.rb [hunk 3]
```diff
@@ -9,9 +9,9 @@
 User.delete_all
 Item.delete_all
 
-User.create!(name: "Brian Casey", email: "email@email.com", payment: "venmo at bcasey13")
-User.create!(name: "Lady Gaga", email: "gaga@email.com", payment: "cash")
-User.create!(name: "Meg Imperato", email: "mimperato@colgate.edu", payment: "cash")
+User.create!(name: "Brian Casey", email: "email@email.com", payment: "venmo at bcasey13", password: "gogate")
+User.create!(name: "Lady Gaga", email: "gaga@email.com", payment: "cash", password: "rarara")
```
Comment (amycsong98) [inline marketplaceApp/db/seeds.rb:13]: LOL

**Review by amycsong98 — APPROVED** [review]
LGTM

File: marketplaceApp/app/controllers/users_controller.rb (hunk 1 above)
Comment (Allegraknox) [reply marketplaceApp/app/controllers/users_controller.rb:12]: thanks I will update before merging

File: marketplaceApp/app/views/users/registrations/edit.html.erb (hunk 2 above)
Comment (Allegraknox) [reply marketplaceApp/app/views/users/registrations/edit.html.erb:49]: Will update before merging, good point

**Pull request merged**